### PR TITLE
Give unique id(eventid) for each handler request to tight with action instance

### PR DESCRIPTION
**Pull request by lixingwang**
**What kind of change does this PR introduce?** (check one with "x")
```
[] Bugfix
[*] Feature
[] Code style update (formatting, local variables)
[] Refactoring (no functional changes, no api changes)
[] Other... Please describe:
```

**Fixes**: #252 

**What is the current behavior?**
* Today there is no way that we can track requests from triggers always down to action instances in logs.  such as Trigger -> Handler -> Action -> Done.  It is super difficult to debugging from logs or find a way that can associate one request from handler to flow instance.  
* No handler execution time took been printed at logs
**What is the new behavior?**

* I came up with a way that each request from the hander will associate with a unique key(Event Id with UUID), the key will be pass to action instance through context.  From the action side, they can get the key from context and use it for logging for tracking purposes. 
* The Event Id should able to be overridden by trigger developer where they can give their business logic perspective id.  such as Message-ID etc.
* Print time took by handle each handler. 

Example logs:

image

### CODE REVIEW

File: trigger/context.go [hunk 1]
```diff
@@ -2,26 +2,60 @@ package trigger
...
 	"context"
+	"github.com/project-flogo/core/support"
+	"time"
 )
 
-type key int
+var handlerKey = "_ctx_handler_info"
 
-var handlerKey key
+var idGenerator, _ = support.NewGenerator()
```
Comment (mellistibco) [inline trigger/context.go:11]: need to make sure newgen hasn't failed...

**Review by mellistibco — CHANGES_REQUESTED** [review]

File: trigger/context.go (hunk 1 above)
Comment (lixingwang) [reply trigger/context.go]: Good catch @mellistibco   Fixed it with just panic,  as it is very unlikely to happens

**Review by mellistibco — APPROVED** [review]

File: trigger/context.go (hunk 1 above)
Comment (vnalawad-tibco) [reply trigger/context.go]: @lixingwang @mellistibco Should that be a panic? Can we probably retry or use timestamp instead in such failure?

**Review by vnalawad-tibco — APPROVED** [review]

File: trigger/context.go (hunk 1 above)
Comment (lixingwang) [reply trigger/context.go]: init step should be fine.  If the error happens it gonna be a big issue.

**Pull request merged**